### ISSUE TITLE
dotnet-restore hangs indefinitely

### ISSUE THREAD

**Issue by natemcmaster** [NuGet/Home#2371]
@aspnet CI agents have suffered from periodic builds in which dotnet-restore hangs indefinitely. The process has to be manually killed. This appears to occur randomly. The hang happens part-way through a restore operation.

The most recent hang occurred earlier today. I gathered a process dump and the restore context. Contact me internally so I can send you the dump (too big to attach here).

Using dotnet-cli 1.0.0.001718
Running Win Server 2012
Restore feeds: myget.org/F/aspnetcidev (v3), nuget.org (v3)

**Comment by emgarten** [comment]
This is due to an azure storage problem. There is a fix in progress currently to help with bad sources that should help with this: https://github.com/NuGet/NuGet.Client/pull/390

**Comment by natemcmaster** [comment]
cc @pranavkm @victorhurdugaci

**Comment by yishaigalatzer** [comment]
@natemcmaster - Please try a new CLI with the latest dev build, it should be resolved now

CC @piotrpMSFT

**Comment by joelverhagen** [comment]
This is fixed with a change to add timeout to downloads (fail is no bytes received from the network in X seconds).
Duplicate of https://github.com/NuGet/Home/issues/2269

**Comment by natemcmaster** [comment]
Thanks. We are not able to upgrade dotnet-cli due to other blocking bugs in dotnet-cli. Any workaround in the meantime? We have had to manually kill dotnet-restore for the last dozen or so runs of EF on aspnetci.

**Comment by TheRealPiotrP** [comment]
What are you blocked on?

Sent using OWA for iPhone

**Comment by natemcmaster** [comment]
@piotrpMSFT https://github.com/dotnet/cli/issues/1905, but the fix supposedly went in last night. We'll try again today
